src/modules/user/domain/user.service.ts

```diff
@@ -0,0 +1,38 @@
+import { ClassProvider, Inject, Injectable } from "@nestjs/common";
+import {
+	ServiceCreateUserInput,
+	ServiceUser,
+	UserServiceDefinition,
+	UserServiceName,
+} from "src/modules/user/domain/user.service.types";
+import {
+	RepoUser,
+	UserRepositoryDefinition,
+	UserRepositoryName,
+} from "src/modules/user/repository/user.repository.types";
+
+@Injectable()
+export class UserService implements UserServiceDefinition {
+	constructor(@Inject(UserRepositoryName) private readonly userRepository: UserRepositoryDefinition) {}
```

Comment:
This is where I've defined the repository we'll be using in our Service. So if you want to add more you can. For now all we need is the `userRepository`